### PR TITLE
Inset the full screen modal used for workspaces on large devices

### PR DESCRIPTION
**Pull request by HorusGoul**
cc @marcaaron 

### Details
Added the expected spacing around the full-screen modals without modifying too many things. 

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/169135

### Tests / QA Steps

Using a large screen (like the web version or the desktop app):
1. Create a new workspace
2. Go to its settings page
3. Verify that the full-screen modal is separated from the window border and that its corners are rounded, like this photo:
image

Using a small screen (mobile version):
1. Create a new workspace
2. Go to its settings page
3. Verify that the full-screen modal is not separated from the window border and has no rounded corners, like in this photo:
image

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web
image

#### Mobile Web
image

#### Desktop
image

#### iOS
image

#### Android
image

### CODE REVIEW

**Comment by HorusGoul** [comment]
Should we set the overlay background color to the one used in the Figma design or leave it as `rgba(0, 0, 0, 0.5)`?

**Comment by shawnborton** [comment]
I think we should already have a color variable for the correct overlay color. If you launch an attachment modal from a chat, you should see what I mean.

**Comment by HorusGoul** [comment]
Done! I had to create a new component to customize the background color. Let me know what you think about the changes.

**Comment by shawnborton** [comment]
Mind sharing a screenshot?

**Comment by marcaaron** [comment]
I'm not sure if this is a blocker, but interacting with the backdrop won't dismiss this modal.

File: src/components/CardOverlay.js
Comment (marcaaron) [inline src/components/CardOverlay.js:14]: This style should be moved into `styles.js`. We have standardized on avoiding inline styles and there is some more explanation [here](https://github.com/Expensify/Expensify.cash/blob/main/STYLING.md#inline-styles).

**Review by marcaaron — CHANGES_REQUESTED** [review]
Mostly LGTM! But did have some questions

**Comment by HorusGoul** [comment]
> Mind sharing a screenshot?

image

File: src/libs/Navigation/AppNavigator/BaseDrawerNavigator.js
Comment (HorusGoul) [reply src/libs/Navigation/AppNavigator/BaseDrawerNavigator.js:60]: That's what I tried first, and I found that the sceneContainerStyle targets a different UI component. 

Adding the View wrapper was necessary because it was the only way to wrap the Drawer and the Screen contents and apply the rounded corners to them.

File: src/styles/styles.js
Comment (HorusGoul) [reply src/styles/styles.js:1862]: Yep, the non-relative values were causing issues after I added the spacing around the modal (essentially, the modal wasn't respecting the parent's right and bottom padding), so I ended up making all the sizes relative to fix the issue.

An alternative to this was subtracting `spacing size * 2` from the absolute values, but that was a worse solution IMO.

File: src/components/CardOverlay.js
Comment (HorusGoul) [reply src/components/CardOverlay.js]: Done!

**Comment by HorusGoul** [comment]
@marcaaron

> I'm not sure if this is a blocker, but interacting with the backdrop won't dismiss this modal.

We could use a Touchable inside the CardOverlay to handle these interactions. However, the visible surface of the overlay is tiny (20px per side), so it may not add that much value. 

Have we considered adding support for the `Esc` key to close modals? I think that would be more suited for this use case.

**Comment by marcaaron** [comment]
>We could use a Touchable inside the CardOverlay to handle these interactions. However, the visible surface of the overlay is tiny (20px per side), so it may not add that much value.

I'll defer to @shawnborton on this one. I see your point about the small surface, but also can see an argument where the universal expected behavior for a backdrop behind a modal (however small) would be to dismiss the modal. No strong feelings on it though :)

>Have we considered adding support for the Esc key to close modals? I think that would be more suited for this use case.

`esc` key should already be enabled, yes

File: src/libs/Navigation/AppNavigator/BaseDrawerNavigator.js
Comment (marcaaron) [reply src/libs/Navigation/AppNavigator/BaseDrawerNavigator.js:60]: Got it. Does having the extra `View` around everything else cause issues as well? Something like

```js
<View style={props.isSmallScreenWidth ? styles.navigationSceneFullScreenWrapper : undefined}>
     ... other stuff
```

Comment (marcaaron) [reply src/libs/Navigation/AppNavigator/BaseDrawerNavigator.js:60]: Also no worries if there is no way to keep all the JSX together. I think it's just easier to parse without the `content` variable.

File: src/styles/styles.js
Comment (marcaaron) [reply src/styles/styles.js:1872]: >Yep, the non-relative values were causing issues after I added the spacing around the modal (essentially, the modal wasn't respecting the parent's right and bottom padding), so I ended up making all the sizes relative to fix the issue.

Got it, thanks. I think maybe in this case it would be good to add that context in a comment since a future change could undo that and the reasoning for these values is not very clear.

>An alternative to this was subtracting spacing size * 2 from the absolute values, but that was a worse solution IMO.

That does sound more complicated.

**Review by marcaaron — CHANGES_REQUESTED** [review]
Code changes look good, but I think this PR is introducing a regression as the height expands much larger than the screen size. See video:

https://user-images.githubusercontent.com/32969087/124803570-fcd2ec00-def4-11eb-8970-db719ab6f551.mp4

**Comment by shawnborton** [comment]
I don't feel strongly about the outside area being clickable. I agree with Marc though that it is expected behavior, but I could live without it here. I feel like the `Esc` key is more useful though so if I had to pick, I would want the `Esc` key before the outer area.

**Comment by HorusGoul** [comment]
I'll pick this back after coming back from OOO 😄

File: src/styles/styles.js
Comment (joelbettner) [inline src/styles/styles.js:1439]: I'm pretty sure you mean to name this with "Screen" and not "Scene"...correct?

**Review by joelbettner — CHANGES_REQUESTED** [review]
Left one comment. Other than that it LGTM.

File: src/libs/Navigation/AppNavigator/AuthScreens.js
Comment (joelbettner) [inline src/libs/Navigation/AppNavigator/AuthScreens.js:221]: Since `styles.cardOverlay` is being added below and has the following styles:
```js
    cardOverlay: {
        backgroundColor: themeColors.modalBackdrop,
        position: 'absolute',
        top: 0,
        left: 0,
        width: '100%',
        height: '100%',
        opacity: 0.5,
    }
```

and `styles.fullscreenCard` has the following styles:
```js
    fullscreenCard: {
        position: 'absolute',
        left: 0,
        top: 0,
        width: '100%',
        height: '100%',
    }
```

why do we need to add `...styles.fullscreenCard` here? Won't all these styles be included in `cardOverlay`?

File: src/libs/Navigation/AppNavigator/AuthScreens.js
Comment (joelbettner) [inline src/libs/Navigation/AppNavigator/AuthScreens.js:227]: Sould it make more sense just to have

```js
cardOverlay: <View style={styles.cardOverlay} />,
```

Why have an entirely different file for something that is only used in this one place?

**Review by joelbettner — CHANGES_REQUESTED** [review]

File: src/libs/Navigation/AppNavigator/AuthScreens.js
Comment (HorusGoul) [reply src/libs/Navigation/AppNavigator/AuthScreens.js:245]: The `cardOverlay` is for the backdrop of the modal, and the `fullscreenCard` is for the content that appears on top of the backdrop. These styles are for two different components, hence the need for the two different styles.

File: src/libs/Navigation/AppNavigator/AuthScreens.js
Comment (HorusGoul) [reply src/libs/Navigation/AppNavigator/AuthScreens.js:251]: We could move it to the AuthScreen but, we would still need to pass it as `cardOverlay: CardOverlay`. I think it's best to leave it as a separate file just in case we need to use it for another modal.

**Comment by HorusGoul** [comment]
I just fixed the bug @marcaaron mentioned for the Reports screen. I still need to test other screens similar to the Reports one to see if there are similar regressions.

**Comment by marcaaron** [comment]
Let us know when it's ready for another review 👍

**Comment by HorusGoul** [comment]
Ready for review!

**Comment by HorusGoul** [comment]
I have read the CLA Document and I hereby sign the CLA

File: src/libs/Navigation/AppNavigator/AuthScreens.js
Comment (joelbettner) [reply src/libs/Navigation/AppNavigator/AuthScreens.js:245]: I was just wondering if the styles would be inherited. In any case, if the `cardOverlay` component might be used elsewhere, it makes sense to pass the styles along and not rely on inheritence.

**Review by joelbettner — DISMISSED** [review]

**Comment by marcaaron** [comment]
I'm not sure if this PR was blocked on me (sorry, if so) but it has conflicts now.

**Comment by HorusGoul** [comment]
Conflicts fixed. If you could review @marcaaron

**Review by joelbettner — APPROVED** [review]

**Review by marcaaron — APPROVED** [review]

**Pull request merged**
